File: legacy/src/main/java/org/ubdev/task/_3005/solution.cpp
#include <vector>

class Solution {
public:
  int maxFrequencyElements(std::vector<int> &nums) {
    int max = 0;
    int count = 0;
    int map[101];
    for (int num : nums) {
      map[num]++;
      if (map[num] > max) {
        count = 1;
        max = map[num];
      } else if (map[num] == max) {
        count++;
      }
    }
    return max * count;
  }
};

File: legacy/src/main/java/org/ubdev/task/_3005/solution.rs
impl Solution {
    pub fn max_frequency_elements_my(nums: Vec<i32>) -> i32 {
        let vec: Vec<i32> = nums.iter()
            .fold(vec![0; 101], |mut map, n| {map[*n as usize] += 1; map});
        let max = vec.iter()
            .max()
            .unwrap();
        max * vec.iter().filter(|x| *x == max).count() as i32
    }

    pub fn max_frequency_elements(mut n: Vec<i32>) -> i32 {
        n.sort_unstable(); n.chunk_by(|a, b| a == b)
        .fold((0, 0, 0), |r, c| if c.len() > r.0 { (c.len(), 1, c.len())} 
            else if c.len() == r.0 { (r.0, r.1 + 1, r.0 * (r.1+1))} else { r }).2 as _
    }
}
